flycheck.el

```diff
@@ -3610,6 +3663,40 @@ list."
       ;; highlighting
       (flycheck-error-list-highlight-errors preserve-pos))))
 
+(defun flycheck-error-list-mode-line-filter-indicator ()
+  "Create a string representing the current error list filter."
+  (if flycheck-error-list-minimum-level
+      (format " [>= %s]" (symbol-name flycheck-error-list-minimum-level))
+    ""))
+
+(defun flycheck-error-list-set-filter (level)
+  "Restrict the error list to errors at level LEVEL or higher."
+  (interactive
+   (list (read-flycheck-error-level
+          "Minimum error level (errors at lower levels will be hidden): ")))
+  (-when-let (buf (get-buffer flycheck-error-list-buffer))
+    (with-current-buffer buf
+      (set (make-local-variable 'flycheck-error-list-minimum-level)
+           (and (flycheck-error-level-p level) level)))
```

Comment:
I think we should not silently reset any filter if the `level` is invalid, but instead signal an error.   The current behaviour will be confusing in my opinion, because users can easily reset filters accidentally:  As far as I can see `read-flycheck-error-level` does not require matching input and returns whatever symbol the user enters, which in turn will pass an invalid `level` to this function, causing the `and` clause to return `nil` and thus resetting the filter.